src/components/shared/generalDropdown.tsx

```diff
@@ -0,0 +1,32 @@
+import Dropdown from 'react-dropdown';
+import '../../styles/generalDropdown.scss';
+import '../../styles/operatorDropdown.scss';
+
+interface GeneralDropdownProps {
+  onChange: (option: string) => void;
+  options: string[];
+  position: string;
+  className?: string;
+}
+
+export function GeneralDropdown(props:GeneralDropdownProps):JSX.Element  {
+  const options: {value:string, label:string, className:string}[] = [];
+  props.options.map((option) => {
+    if (option != props.options[props.options.length-1]) {
+      options.push({value: option, label: option, className: 'option'});
+    }
+    else {
+      options.push({value: option, label: option, className: 'option last'});
+    }
+  });
```

Comment:
```suggestion
  const options: {value:string, label:string, className:string}[] = props.options.map((option) => {
    const opt = {value: option, label: option, className: 'option'};
    if (option === props.options[props.options.length-1]) {
      opt.className = 'option last';
    } 
    return opt;
  });
```
options.map should be moved into the initialization step (better not to have function calls inside JSX element body)